### ISSUE TITLE
[inductor] torch.compile's RNG state is inconsistent with eager on non-contiguous tensor when size >= 16

### ISSUE THREAD

**Issue by maybeLee** [pytorch/pytorch#177652]
### 🐛 Describe the bug

When compiling a function with `torch.randn_like`, the compiled function's return result is inconsistent with eager even when the random seed is fixed and set `torch._inductor.config.fallback_random = True`.
This issue only happens when the input of the function is a non-contiguous tensor (created via `permute(1,0)`), which size larger than 16 and `shape[0], shape[1]!=1`.

Here is the code to reproduce:

```python
import torch
import itertools
torch._inductor.config.fallback_random = True

def fn(x):
    torch.manual_seed(0)
    return torch.randn_like(x)

# shape[0] * shape[1] >= 16 and shape[0], shape[1] != 1
for shape in itertools.product(range(1, 17), range(1, 17)):
    shape = shape
    x1 = torch.zeros(shape).permute(1, 0)
    x2 = torch.zeros(shape).permute(1, 0)
    
    cfunc = torch.compile(fn, backend='inductor')
    
    torch.manual_seed(0)
    out1 = fn(x1)
    # print(f'{out1=}')
    
    torch.manual_seed(0)
    out2 = fn(x1)
    # print(f'{out2=}')
    
    torch.manual_seed(0)
    out3 = cfunc(x2)
    # print(f'{out3=}')
    
    torch.manual_seed(0)
    out4 = cfunc(x2)
    # print(f'{out4=}')
    
    try:
        torch.testing.assert_close(out1, out3, equal_nan=True)
    except AssertionError as e:
        print(f'Shape {shape} will cause inconsistency.')

```

And the output shows that only specific shape can trigger this inconsistency:

```
Shape (2, 8) will cause inconsistency.
Shape (2, 9) will cause inconsistency.
Shape (2, 10) will cause inconsistency.
Shape (2, 11) will cause inconsistency.
Shape (2, 12) will cause inconsistency.
Shape (2, 13) will cause inconsistency.
Shape (2, 14) will cause inconsistency.
Shape (2, 15) will cause inconsistency.
Shape (2, 16) will cause inconsistency.
Shape (3, 6) will cause inconsistency.
Shape (3, 7) will cause inconsistency.
Shape (3, 8) will cause inconsistency.
Shape (3, 9) will cause inconsistency.
...
```

Note that this issue does not happen on backend `eager` and `aot_eager`.
This issue also does not happen when running on cuda (RTX 6000).

### Versions

Versions of relevant libraries:
[pip3] numpy==2.4.2
[pip3] torch==2.12.0.dev20260316+cpu
[pip3] torchvision==0.26.0.dev20260223+cpu
[pip3] triton==3.6.0
[conda] numpy 2.4.2 pypi_0 pypi
[conda] torch 2.12.0.dev20260316+cpu pypi_0 pypi
[conda] torchvision 0.26.0.dev20260223+cpu pypi_0 pypi
[conda] triton 3.6.0 pypi_0 pypi

cc @chauhang @penguinwu
